https://github.com/emscripten-ports/SDL2/pull/65
Remove when emscripten-ports/SDL2 is tagged version_16 and grabbed by emcc

commit 86308101a27a51272a2f57de46c6afa7013f3a74
Author: Charlie Birks <charlie@daftgames.net>
Date:   Sat Dec 8 12:38:48 2018 +0000

    Emscripten: reset fullscreen_window when leaving fullscreen
    
    If the browser left fullscreen mode by the user pressing ESC, the next
    call to SDL_SetWindowFullscreen(1) will fail as it thinks the window is
    already fullscreen. (#65)

diff --git a/src/video/emscripten/SDL_emscriptenevents.c b/src/video/emscripten/SDL_emscriptenevents.c
index 4ba8453ec..b2981dd82 100644
--- a/src/video/emscripten/SDL_emscriptenevents.c
+++ b/src/video/emscripten/SDL_emscriptenevents.c
@@ -551,6 +551,8 @@ static EM_BOOL
 Emscripten_HandleFullscreenChange(int eventType, const EmscriptenFullscreenChangeEvent *fullscreenChangeEvent, void *userData)
 {
     SDL_WindowData *window_data = userData;
+    SDL_VideoDisplay *display;
+
     if(fullscreenChangeEvent->isFullscreen)
     {
         window_data->window->flags |= window_data->requested_fullscreen_mode;
@@ -563,6 +565,13 @@ Emscripten_HandleFullscreenChange(int eventType, const EmscriptenFullscreenChang
     else
     {
         window_data->window->flags &= ~FULLSCREEN_MASK;
+
+        /* reset fullscreen window if the browser left fullscreen */
+        display = SDL_GetDisplayForWindow(window_data->window);
+
+        if (display->fullscreen_window == window_data->window) {
+            display->fullscreen_window = NULL;
+        }
     }
 
     return 0;
